File: app/src/main/java/com/example/stas/oxforddictionary/presentation/view/entry/EntryContract.java
package com.example.stas.oxforddictionary.presentation.view.entry;


import com.example.stas.oxforddictionary.presentation.view.entry.adapter.Item;
import com.example.stas.oxforddictionary.presentation.view.base.CanShowError;

import java.util.List;

public interface EntryContract {
    interface View extends CanShowError {
        void showDefinition(List<Item> definitions, List<String> titleSet);
        void playSound(String soundURL);
        void hideProgressBar();
    }
    interface Presenter{
        void detachView();
        void attachView(EntryContract.View view);
        void getDefinition(String word);
        void getSound(String word);


    }
}

File: app/src/main/java/com/example/stas/oxforddictionary/presentation/view/entry/EntryContract.kt
package com.example.stas.oxforddictionary.presentation.view.entry

import com.example.stas.oxforddictionary.data.database.model.ViewedWordModel
import com.example.stas.oxforddictionary.presentation.view.base.CanShowError
import com.example.stas.oxforddictionary.presentation.view.entry.adapter.Item
import io.reactivex.Flowable
import io.reactivex.Observable

interface EntryContract {
    interface View : CanShowError {
        fun showDefinition(definitions: List<Item>, titleSet: List<String?>)
        fun playSound(soundURL: String?)
        fun hideProgressBar()
       fun showAutocompletes(t: List<String>?)
    }

    interface Presenter {
        fun detachView()
        fun attachView(view: EntryContract.View)
        fun getDefinition(word: String)
        fun getSound(word: String)
        fun saveDefinition(word: String, definition: String)
        fun onEntryTextChanged(text: String): Observable<List<ViewedWordModel>>
    }
}
